base/task.jl

```diff
@@ -5,7 +5,56 @@
 # allow tasks to be constructed with arbitrary function objects
 Task(f) = Task(()->f())
 
-show(io::IO, t::Task) = print(io, "Task ($(t.state)) @0x$(hex(convert(UInt, pointer_from_objref(t)), WORD_SIZE>>2))")
+function show(io::IO, t::Task)
+    print(io, "Task ($(t.state)) @0x$(hex(convert(UInt, pointer_from_objref(t)), WORD_SIZE>>2))")
+    if t.state == :failed
+        show(io, CapturedException(t.result, t.backtrace))
+    end
+end
+
+type CapturedException
```

Comment:
> I am not really clear what decides if an object should be a subtype of `Exception`.

Not sure who would know the answer to that question, though one difference I've noticed is if you want FactCheck to catch an error it works much nicer with things are subtypes of `Exception` than things that aren't.